cfme/services/catalogs/catalog.py

```diff
@@ -1,115 +1,121 @@
-
-from cfme import web_ui
-from cfme.exceptions import DestinationNotFound
-from cfme.fixtures import pytest_selenium as sel
-from cfme.web_ui import accordion, flash, form_buttons, Input
-from cfme.web_ui import toolbar as tb, paginator, CheckboxTable, match_location
-from utils.appliance import Navigatable
-from utils.appliance.implementations.ui import navigator, navigate_to, CFMENavigateStep
-from utils.blockers import BZ
-from utils.pretty import Pretty
 from utils.update import Updateable
+from utils.pretty import Pretty
+from utils.appliance import Navigatable
+from utils.appliance.implementations.ui import navigator, CFMENavigateStep, navigate_to
 from utils import version
 
 
-cfg_btn = partial(tb.select, "Configuration")
-catalog_tree = partial(accordion.tree, "Catalogs")
+from . import ServicesCatalogView
+
+
+class CatalogForm(ServicesCatalogView):
+    title = Text('#explorer_title_text')
+
+    name = Input(name='name')
+    description = Input(name="description")
+
+    save_button = Button('Save')
+    cancel_button = Button('Cancel')
+
+
+class CatalogsView(ServicesCatalogView):
+    title = Text("#explorer_title_text")
+
+    @property
+    def is_displayed(self):
+        return (
+            self.in_explorer and
+            self.title.text == 'All Catalogs' and
+            self.catalogs.is_opened and
+            self.catalogs.tree.currently_selected == ["All Catalogs"])
+
+
+class DetailsCatalogView(ServicesCatalogView):
+    title = Text("#explorer_title_text")
+
+    @property
+    def is_displayed(self):
+        return (
+            self.in_explorer and
+            self.title.text == 'Catalog "{}"'.format(self.context['object'].name) and
+            self.catalogs.is_opened
+        )
 
-listview_table = CheckboxTable(table_locator='//div[@id="list_grid"]/table')
 
-item_multiselect = web_ui.MultiSelect(
-    "//select[@id='available_fields']",
-    "//select[@id='selected_fields']",
-    "//div[@id='column_lists']//a[contains(@href, 'button=right')]/img",
-    "//div[@id='column_lists']//a[contains(@href, 'button=left')]/img")
+class AddCatalogView(CatalogForm):
 
-form = web_ui.Form(
-    fields=[('name_text', Input("name")),
-            ('description_text', Input("description")),
-            ('button_multiselect', item_multiselect),
-            ('add_button', form_buttons.add),
-            ('save_button', form_buttons.save),
-            ('cancel_button', form_buttons.cancel)])
+    add_button = Button("Add")
 
-item_form = web_ui.Form(
-    fields=[('type_select', "//select[@id='st_prov_type']"),
-            ('name_text', Input("name")),
-            ('description_text', Input("description")),
-            ('display_checkbox', Input("display")),
-            ('add_button', form_buttons.add)])
+    @property
+    def is_displayed(self):
+        return (
+            self.title.text == "Adding a new Catalog" and
+            self.catalogs.is_opened
```

Comment:
so, it should also check in_explorer. And the is_opened should come before the title text. So the order then would be in_explorer, catalogs, title text.